scripts/lib/CIME/SystemTests/system_tests_compare_two.py

```diff
@@ -92,7 +92,10 @@ def __init__(self,
         self._case2 = None
 
         self._setup_cases_if_not_yet_done()
-
+        # Since case 2 has the same name as case1 its CIME_OUTPUT_ROOT must also be different
+        self._case2.set_value("CIME_OUTPUT_ROOT",
```

Comment:
This should go in the _setup_cases function, in order to keep the __init__ method cleaner and so that any problem is protected by the try... except block in _setup_cases_if_not_yet_done.